sootup.callgraph/src/main/java/sootup/callgraph/GraphBasedCallGraph.java

```diff
@@ -129,6 +133,93 @@ public int callCount() {
     return graph.edgeSet().size();
   }
 
+  @Override
+  public String exportAsDot() {
+    StringBuilder dotFormatBuilder = new StringBuilder();
+
+    graph.edgeSet().stream()
+        .sorted(
+            Comparator.comparing(
+                    (Edge edge) -> graph.getEdgeSource(edge).toString(), Comparator.naturalOrder())
+                .thenComparing(
+                    (Edge edge) -> graph.getEdgeTarget(edge).toString(), Comparator.naturalOrder()))
+        .forEach(
+            edge -> {
+              Vertex sourceVertex = graph.getEdgeSource(edge);
+              Vertex targetVertex = graph.getEdgeTarget(edge);
+              dotFormatBuilder
+                  .append("\t")
+                  .append("\"" + sourceVertex.methodSignature + "\"")
+                  .append(" -> ")
+                  .append("\"" + targetVertex.methodSignature + "\"")
+                  .append(";\n");
+            });
+    // Sort the callGraph
+    String sortedCallGraph = sortCallGraph(dotFormatBuilder.toString());
+
+    // Using only one string builder, so deleting everything and adding the sorted callgraph
+    dotFormatBuilder.delete(0, dotFormatBuilder.capacity());
```

Comment:
Don't do this. 
Instead just return the string concatination 
"strict digraph ObjectGraph {\n" + sortedCallgraph + "}"

Reply:
deleting the string builder seems unnecessary complex, and it might waste execution time